src/Microsoft.OData.Edm/Schema/EdmOperation.cs

```diff
@@ -83,20 +83,28 @@ public string FullName
         }
 
         /// <summary>
-        /// Gets the return type of this function.
+        /// Gets the return type of this operation.
         /// </summary>
-        public IEdmTypeReference ReturnType { get; private set; }
+        public IEdmTypeReference ReturnType
+        {
+            get { return Return == null ? null : Return.Type; }
+        }
 
         /// <summary>
-        /// Gets the parameters of this function.
+        /// Gets the parameters of this operation.
         /// </summary>
         public IEnumerable<IEdmOperationParameter> Parameters
         {
             get { return this.parameters; }
         }
 
         /// <summary>
-        /// Searches for a parameter with the given name in this function and returns null if no such parameter exists.
+        /// Gets the return of this operation.
+        /// </summary>
+        internal IEdmOperationReturn Return { get; private set; }

```

Comment:
>interna [](start = 8, length = 7)

Might we want to make this public? If not, how would someone get/add annotations to the operation return?

Reply:
Call the extension method to add. See the CsdlWriterTest.

Reply:
Since the extension method is public, that works.  Thanks.

---
In reply to: [262675031](https://github.com/OData/odata.net/pull/1414#discussion_r262675031) [](ancestors = 262675031)